hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -454,9 +456,8 @@ private object HoodieMergeOnReadRDD {
   }
 
   private def registerStructTypeSerializerIfNeed(schemas: List[StructType]): Unit = {
```

Comment:
Why do we need this method? 
Schema registration should be adhoc -- whenever someone is requesting to serialize `HoodieSparkRecord` we'd cache the schema, replacing w/ the fingerprint

Reply:
Removed.